lib/Common/Memory/Recycler.h

```diff
@@ -2372,7 +2375,8 @@ struct ForceLeafAllocator<RecyclerNonLeafAllocator>
     typedef RecyclerLeafAllocator AllocatorType;
 };
 
-#ifdef PROFILE_EXEC
+// TODO: enable -profile for GC phases
+#if defined(PROFILE_EXEC) && 0
```

Comment:
Background GC threads use the one profiler from the same recycler object. Profiler object is not thread safe (phase push/pop cannot match if accessed from multiple threads).

---
In reply to: [66281159](https://github.com/Microsoft/ChakraCore/pull/1096#discussion_r66281159) [](ancestors = 66281159)